src/v/kafka/server/group.cc

```diff
@@ -3229,6 +3233,12 @@ void group::maybe_rearm_timer() {
 }
 
 ss::future<> group::do_abort_old_txes() {
+    if (!_catchup_lock->try_read_lock()) {
+        co_return;
+    }
+    _catchup_lock->read_unlock();
+
+    auto units = co_await _catchup_lock->hold_read_lock();
```

Comment:
I think there is still a scope for deadlock here because of co_await. A co_await means there could be a scheduling point after L3240 (we are at the mercy of the scheduler). We should get rid of co_await to be 100% safe?

Reply:
Also the behavior is subtle, so that could use a comment.

Reply:
semaphore has a units-based try-lock interface. i propose we add a similar interface to seastar's rwlock. i don't think this is the only place in the tree where we have this pattern.

Reply:
added

Reply:
the interface was added but it isn't used anywhere

Reply:
Shouldn't this use the new ssx::rwlock?